contracts/multisigs/GnosisSafeMultisig.sol

```diff
@@ -51,14 +65,20 @@ contract GnosisSafeMultisig {
                 offset := add(offset, 20)
                 paymentReceiver := mload(add(data, offset))
 
-                // Read all the uints
+                // Read all the uints (64 more bytes, a total of 144 bytes)
                 offset := add(offset, 32)
                 payment := mload(add(data, offset))
                 offset := add(offset, 32)
                 nonce := mload(add(data, offset))
+            }
 
-                // Read the payload data
-                payload := mload(add(data, dataSize))
+            // Read the payload, if provided
+            if (dataLength > DEFAULT_DATA_LENGTH) {
+                uint256 payloadLength = dataLength - DEFAULT_DATA_LENGTH;
+                payload = new bytes(payloadLength);
+                for (uint256 i = 0; i < payloadLength; ++i) {
+                    payload[i] = data[i + DEFAULT_DATA_LENGTH];
+                }
```

Comment:
Extracting `bytes` payload in assembly is not that straightforward, it is more convenient and user-friendly to do it in plain solidity

Reply:
What was the error on the lhs?

Reply:
The allocation (line 78) is needed before copying the bytes themselves